src/main/java/io/github/moulberry/notenoughupdates/profileviewer/level/task/MiscTaskLevel.java

```diff
@@ -166,21 +166,38 @@ public void drawTask(JsonObject object, int mouseX, int mouseY, int guiLeft, int
 		}
 
 		int sbXpTimeCharm = 0;
-		if (object.has("rift") &&
-			object.getAsJsonObject("rift").has("gallery") &&
-			object.getAsJsonObject("rift").getAsJsonObject("gallery").has("secured_trophies")) {
-			JsonArray timeCharms = object.getAsJsonObject("rift").getAsJsonObject("gallery").getAsJsonArray(
-				"secured_trophies");
-			sbXpTimeCharm += timeCharms.size() * miscellaneousTask.get("timecharm_xp").getAsInt();
+		int sbXpBurger = 0;
+		if (object.has("rift")) {
+			JsonObject rift = object.getAsJsonObject("rift");
+			if (rift.has("gallery") &&
+				rift.getAsJsonObject("gallery").has("secured_trophies")) {
+				JsonArray timeCharms = rift.getAsJsonObject("gallery").getAsJsonArray(
+					"secured_trophies");
+				sbXpTimeCharm += timeCharms.size() * miscellaneousTask.get("timecharm_xp").getAsInt();
+			}
+
+
+			if (rift.has("castle") && rift.getAsJsonObject("castle").has("grubber_stacks") && miscellaneousTask.has("mcgrubber_burger_xp")) {
+				sbXpBurger = miscellaneousTask.get("mcgrubber_burger_xp").getAsInt()
```

Comment:
No Utils.getElement?

Reply:
if i changed that now i would have to do it for all of them and i rly cba ;-;

Reply:
not really in the mood to approve something that crashes, tbh

Reply:
Y would it crash

Reply:
in case `mcgrubber_burger_xp` does not exist in that json

Reply:
i have an if check for 'mcgrubber_burger_xp' though